warehouse/internal/repo/table_upload.go

```diff
@@ -124,71 +138,55 @@ func (repo *TableUploads) GetByUploadIDAndTableName(ctx context.Context, uploadI
 	LIMIT 1;
 `
 
-	rows, err := repo.db.QueryContext(ctx, query, uploadID, tableName)
-	if err != nil {
-		return model.TableUpload{}, fmt.Errorf("querying table uploads: %w", err)
-	}
+	row := repo.db.QueryRowContext(ctx, query, uploadID, tableName)
 
-	entries, err := repo.parseRows(rows)
-	if err != nil {
-		return model.TableUpload{}, fmt.Errorf("parsing rows: %w", err)
+	var tableUpload model.TableUpload
+	err := scanTableUpload(row.Scan, &tableUpload)
+	if errors.Is(err, sql.ErrNoRows) {
+		return tableUpload, fmt.Errorf("no table upload found with uploadID: %d, tableName: %s", uploadID, tableName)
 	}
-	if len(entries) == 0 {
-		return model.TableUpload{}, fmt.Errorf("no table upload found with uploadID: %d, tableName: %s", uploadID, tableName)
+	if err != nil {
+		return tableUpload, fmt.Errorf("parsing rows: %w", err)
```

Comment:
Same suggestion here (error context).

Reply:
Adding appropriate information about the action we are performing.